deepchecks/tabular/utils/task_inference.py

```diff
@@ -0,0 +1,101 @@
+#
+"""Utils module containing functionalities to infer the task type and possible label classes."""
+
+__all__ = ['get_possible_classes', 'infer_task_type']
+
+from typing import List, Optional
+
+import pandas as pd
+from sklearn.base import ClassifierMixin
+
+from deepchecks import tabular
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.tabular.utils.feature_inference import is_categorical
+from deepchecks.tabular.utils.task_type import TaskType
+from deepchecks.utils.strings import is_string_column
+from deepchecks.utils.typing import BasicModel
+
+
+def get_possible_classes(model: Optional[BasicModel], train_dataset: 'tabular.Dataset',
+                         test_dataset: Optional['tabular.Dataset'] = None, force_multiclass: bool = False) \
+        -> Optional[List]:
+    """Return the list of allowed classes for classification tasks or None for regression.
+
+    Parameters
+    ----------
+    model : BasicModel
+        Model object used in task
+    train_dataset : 'tabular.Dataset'
+        Train Dataset of task
+    test_dataset : Optional['tabular.Dataset'], default = None
+        Test Dataset of task
+    force_multiclass: bool, default = False
+        Whether to disable the auto infer for task type and return all observed label values.
+    Returns
+    -------
+    Optional[List]
+        The list of possible classes for classification tasks or None for regression
+    """
+    if not isinstance(train_dataset, tabular.Dataset) or (test_dataset is not None and
+                                                          not isinstance(test_dataset, tabular.Dataset)):
+        raise DeepchecksValueError('train_dataset and test_dataset must be of type tabular.Dataset')
+
+    if hasattr(model, 'classes_') and len(model.classes_) > 0:
+        return list(model.classes_)
+    if model is not None and not hasattr(model, 'predict_proba') and not force_multiclass:
+        if is_string_column(train_dataset.label_col):
+            raise DeepchecksValueError('Model was identified as a regression model, but label column was '
+                                       'found to contain strings.')
+        if isinstance(model, ClassifierMixin):
+            raise DeepchecksValueError('Model is a sklearn classification model but lacks the predict_proba method. '
+                                       'Please train the model with probability=True.')
+        return None
+
+    observed_labels = list(train_dataset.label_col)
+    if test_dataset is not None:
+        observed_labels += list(test_dataset.label_col)
+
+    if model is not None:
+        observed_labels += list(model.predict(train_dataset.features_columns))
+        if test_dataset is not None:
+            observed_labels += list(model.predict(test_dataset.features_columns))
+        return sorted(set(observed_labels))
+    elif is_categorical(pd.Series(observed_labels), max_categorical_ratio=0.05) or force_multiclass:
```

Comment:
Add comment here for readability explaining that if a model does not exist, we use is_categorical to determine if regression or classification